WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -14,12 +14,12 @@ struct ItemListView: View {
         VStack {
             headerView
             switch viewModel.state {
-            case .empty, .error:
+            case .initialLoading, .empty, .error:
                 // These cases are handled directly in the dashboard, we do not render
                 // a specific view within the ItemListView to handle them
                 EmptyView()
-            case .initialLoading, .loading, .loaded:
-                listView(viewModel.items)
+            case .loading(let items), .loaded(let items):
+                listView(items)
```

Comment:
I don't think it's reasonable to avoid it with our current design, but nor is it a big deal that we use an empty view... 

Even if we made the ItemListView responsible for showing the full screen views as well, which would be nice, it would probably need to do them using a separate view modifier that the Dashboard actually presents the contents of (like `posModal`)